solidity/dashboard/src/hooks/useSubscribeToConnectorEvents.js

```diff
@@ -15,19 +26,71 @@ const useSubscribeToConnectorEvents = () => {
       dispatch({ type: "app/logout" })
     }
 
+    const showChooseWalletModal = (payload) => {
+      dispatch({
+        type: "transactions/transaction_added_to_queue",
+        payload: payload,
+      })
+      openModal(<WalletSelectionModal />, {
+        title: "Select Wallet",
+      })
+    }
+
+    const sendTransactionsFromQueue = (transactions) => {
+      const transactionObjects = transactions.map(
+        (transaction) => transaction.params[0]
+      )
+
+      dispatch({
+        type: "web3/send_raw_transaction_in_sequence",
+        payload: transactionObjects,
+      })
+    }
+
+    const executeTransactionsInQueue = (transactions) => {
+      if (transactions.length > 0) {
+        dispatch({
+          type: "transactions/clear_queue",
+        })
+        sendTransactionsFromQueue(transactions)
+      }
+    }
+
     if (isConnected && connector) {
       dispatch({ type: "app/login", payload: { address: yourAddress } })
       connector.on("accountsChanged", accountChangedHandler)
       connector.once("disconnect", disconnectHandler)
+
+      if (connector.name === WALLETS.EXPLORER_MODE.name) {
+        connector.eventEmitter.on(
+          "chooseWalletAndSendTransaction",
+          showChooseWalletModal
+        )
```

Comment:
If we deal with https://github.com/keep-network/keep-core/pull/2405/files#r606091686 the connector API will be consistent(highly recomended 😆 ) and `if (connector.name === WALLETS.EXPLORER_MODE.name)` will be unnecessary.